client/src/hooks/useHash.ts

```diff
@@ -0,0 +1,32 @@
+import { useMemo, useState, useEffect } from "react";
+
+import { useParams } from "next/navigation";
+
+export const useHash = () => {
+  const getCurrentHash = useMemo(
```

Comment:
Memoizing this might be a bit overkilling, it shouldn't change frequently and the operations performed to get the hash are not expensive in terms of computing.